pkg/landscaper/installations/executions/template/spifftemplate.go

```diff
@@ -219,3 +221,77 @@ func spiffResolveComponent(cd *cdv2.ComponentDescriptor, cdList *cdv2.ComponentD
 		return result.Value(), info, true
 	}
 }
+
+func spiffGenerateImageOverwrite(cd *cdv2.ComponentDescriptor, cdList *cdv2.ComponentDescriptorList) func(arguments []interface{}, binding dynaml.Binding) (interface{}, dynaml.EvaluationInfo, bool) {
+	return func(arguments []interface{}, binding dynaml.Binding) (interface{}, dynaml.EvaluationInfo, bool) {
+		info := dynaml.DefaultInfo()
+
+		internalCd := cd
+		internalComponents := cdList
+
+		if len(arguments) > 2 {
+			panic("Too many arguments for spiffGenerateImageOverwrite")
+		} else if len(arguments) == 1 {
+			data, err := spiffyaml.Marshal(spiffyaml.NewNode(arguments[0], ""))
+			if err != nil {
+				return info.Error(err.Error())
+			}
+
+			internalCd = &cdv2.ComponentDescriptor{}
+			err = yaml.Unmarshal(data, internalCd)
+			if err != nil {
+				return info.Error(err.Error())
+			}
+		} else if len(arguments) == 2 {
+			cdData, err := spiffyaml.Marshal(spiffyaml.NewNode(arguments[0], ""))
+			if err != nil {
+				return info.Error(err.Error())
+			}
+
+			internalCd = &cdv2.ComponentDescriptor{}
+			if err := yaml.Unmarshal(cdData, internalCd); err != nil {
+				return info.Error(err.Error())
+			}
+
+			componentsData, err := spiffyaml.Marshal(spiffyaml.NewNode(arguments[1], ""))
+			if err != nil {
+				return info.Error(err.Error())
+			}
+
+			internalComponents = &cdv2.ComponentDescriptorList{}
+			if err := yaml.Unmarshal(componentsData, internalComponents); err != nil {
+				return info.Error(err.Error())
+			}
```

Comment:
could be simplified as in the go function

Reply:
This approach is more simple but it does not allow to provide only a ComponentDescriptorList and no CD. If this is no valid use case we should take this alternative.

Reply:
Same as for the go template function: code simplified, and behaviour unchanged.